doc/src/explanation/glossary.md

```diff
@@ -0,0 +1,712 @@
+SymPy. For more general Python terms, refer to the [Python
+glossary](https://docs.python.org/3/glossary.html). Mathematical terms are
+only included here if they have a specific meaning in SymPy. For general
+mathematical definitions, refer to other sources such as
+[Wikipedia](https://www.wikipedia.org/) or
+[MathWorld](https://mathworld.wolfram.com/), as well as the references in the
+documentation for the specific SymPy functions.
+
+<!-- Please follow the following style for the glossary:
+
+- Capitalize the names, unless they refer to code.
+- Each definition should be short (no more than a paragraph).
+- Keep the terms in alphabetical order.
+- Italicize the first occurrence of the term in the definition.
+- Use cross-references to refer to full documentation for functions, etc.
+- Cross-reference the first instance of other terms in definitions (see
+  below).
+- Classes and functions should only be listed here if they are core concepts
+  for understanding how to use SymPy. Classes and functions themselves should
+  be documented in their respective docstrings in the modules reference.
+
+-->
+
+<!-- To cross reference a glossary item elsewhere in the documentation use
+{term}`termname` in Markdown or :term:`termname` in RST. Use
+{term}`Text <termname>` or :term:`text <termname>` to make a link with
+different text. -->
+
+```{glossary}
+
+Antiderivative
+
+    An *antiderivative* of a function $f(x)$ with respect to $x$ is a function
+    $F(x)$ such that $\frac{d}{dx}F(x) = f(x).$ It is also sometimes called an
+    "indefinite integral" of $f(x)$, and written as $\int f(x)\,dx.$
+    Antiderivatives in SymPy can be computed with {func}`~.integrate`. Note
+    some sources call this the "primitive" of $f(x)$, but this terminology is
+    not used in SymPy because it is not as universally used as
+    "antiderivative", and because "primitive" has other meanings in
+    mathematics and in {func}`SymPy <sympy.polys.polytools.primitive>`.
+
+`args`
+
+    The *`args`* property of a SymPy {term}`expression` is a tuple of the
+    top-level {term}`subexpressions <subexpression>` used to create it. They
+    are the arguments to the class used to create the expression. The args of
+    any expression can be obtained by the `.args` attribute. For example,
+    `(1 + x*y).args` is `(1, x*y)`, because it equals `Add(1, x*y)`. The `args`
+    together with {term}`func` completely define an expression. It is always
+    possible to walk the {term}`expression tree` and extract any subexpression
+    of a SymPy expression by repeated use of `.args`. Every SymPy expression
+    can be rebuilt exactly with `func` and `args`, that is,
+    `expr.func(*expr.args) == expr` will always be true of any SymPy
+    expression `expr`. The args of an expression may be the empty tuple `()`,
+    meaning the expression is an {term}`atom`.
+
+Assumptions
+
+    *Assumptions* are a set of predicates on a {term}`symbol` or
```

Comment:
There is a long guide that could be linked to here:
https://docs.sympy.org/latest/guides/assumptions.html